hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacitySchedulerConfigValidator.java

```diff
@@ -27,14 +27,26 @@
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.LinkedHashSet;
+import java.util.List;
 import java.util.Set;
 
 public final class CapacitySchedulerConfigValidator {
   private static final Logger LOG = LoggerFactory.getLogger(
           CapacitySchedulerConfigValidator.class);
 
+  private interface QueueChangeValidator {
+    void validate(CSQueue oldQueue, CSQueue newQueue) throws IOException;
+  }
+
+  private static final List<QueueChangeValidator> QUEUE_CHANGE_VALIDATORS = Arrays.asList(
```

Comment:
This is a good idea, I like it, however I think it would be a better to define the validators as static methods instead, because you are basically using them as functions anyway.

Reply:
It was a bit over engineered, yeah. I changed it to 3 static methods, and I call them without a cycle.

Reply:
I actually liked the cycle-based solution. What I meant is that you do not need to explicitly create new class to implement a functional interface (an interface with only one method), but use an existing method or a lambda.